### ISSUE TITLE
Consolidate buttons on the title screen

### ISSUE THREAD

**Issue by Pentarctagon** [wesnoth/wesnoth#8768]
With the addition of the Community button in #8743, there's getting to be too many buttons present on the title screen:
![322064512-00e3e406-c0fc-44ef-94aa-c87e23fadb02](https://github.com/wesnoth/wesnoth/assets/3030250/035c31d9-e685-4c2b-a8bf-1c4a19e5bfa2)

My thoughts for those:
* Campaigns - needs to stay
* Multiplayer - needs to stay
* Load - needs to stay
* Add-ons - needs to stay
* Map Editor - needs to stay
* Preferences - could maybe be moved to the bottom and given a smaller button with a gear icon?
* Achievements - would prefer to keep, just so players see it and know achievements are a thing that exists
* Community - would like to keep to draw players' attention to the community links, but I don't feel strongly about it
* Credits - could be added to a unified About dialog instead of having its own button
* Quit - somewhat wondering if it could have its own smaller X button elsewhere on the screen, but I'm not sure if that would make it too hard to notice

**Comment by Wedge009** [comment]
I was thinking something similar, given that the Tutorial was just merged into Campaigns not too long ago. (Okay, maybe it has been a while, that was #4566.) This would be a sub-issue of #4531, I take it?

I think I probably agree the 'needs to stay' items are the bare minimum. (Hmm, I forgot Achievements was also just introduced.)
* Preferences/settings are nearly universally portrayed by a gear icon, I think.
* For Community, I'm not sure if a [share icon](https://en.wikipedia.org/wiki/Share_icon) would be appropriate or perhaps better represented as something like a [group of people](https://icon-library.com/icon/community-icon-png-4.html). (BTW, these are just examples I found by searching, I'm not advocating any particular icon for this purpose.)
* Moving Quit to an X button in the corner (I'm used to top-right but I don't know if there are other conventions there) also makes sense to me.
* I've often seen About (if Credits will go there) be represented by a ? icon.

**Comment by CelticMinstrel** [comment]
> * I'm used to top-right but I don't know if there are other conventions

Windows convention is top right, Mac convention is top left. I don't know what the Linux condition.

But we can also quit using the standard window decorators, which as far as I know has the same effect, so I'm not sure we even need a quit button. We could make it a circular X button in the corner but hide it entirely in windowed mode.

> * I've often seen About (if Credits will go there) be represented by a ? icon. (The Stardew Valley example cited in

As I've already stated, I think our existing (i) button is fine for "about". A ? is more suitable for help than about.

**Comment by CelticMinstrel** [comment]
I believe it does do essentially the same thing in 1.18. Maybe the internal logic getting to that point is a bit different, but as far as I know, the net result is the same.

**Comment by babaissarkar** [comment]
> KDE Plasma uses top-right, by default.

I'd like to add that it is customizable, and it's not certain that the user will always have maximize and iconify buttons. My own is configured top-left.

![Screenshot from 2024-04-17 08-20-28](https://github.com/wesnoth/wesnoth/assets/8469888/c598d1f5-e0da-4078-ae77-c3d819e9687b)

There's also the option of expanding icons on hover to show more details (makes the UI look more simpler) or actually animating them altogether, as I already mentioned [here](https://github.com/wesnoth/wesnoth/issues/4531#issuecomment-2046491311). though animation might need more code.

I'd also like all icons to stay at once place. Currently some are at top right, the info icon is at bottom left, and the language button is at the bottom right.

**Comment by Wedge009** [comment]
I did say 'by default'. I expected it would be customisable. I'm not at my home PC to check right now (still on Plasma 5.27 anyway).

I thought it would make sense to try to move buttons to corners, etc, if the goal is to reduce the current stack of buttons. That sounds like it might be getting into the broader topic of #4531 though.

**Comment by CelticMinstrel** [comment]
> I'd also like all icons to stay at once place. Currently some are at top right, the info icon is at bottom left, and the language button is at the bottom right. Looks a bit haphazard.

I could be on board with that for the most part, but an X button for quit should go along the top, whereas I don't think there's a need to move the current buttons along the bottom.

Maybe language could be moved to the left, but there might be merit in it being off by itself too.

**Comment by babaissarkar** [comment]
> Maybe language could be moved to the left, but there might be merit in it being off by itself too.
Another suggestion would be to move language and it's dialog inside the Preferences dialog, making it a tab.

> Achievements - would prefer to keep, just so players see it and know achievements are a thing that exists
I wonder if in addition to the global scope, would it be good to have an "Achievements" menu inside the campaign itself, which shows the Achievements from that campaign only.
Most games do dedicate multiple dialogs to Achievements, so it's definitely important enough to be kept in the title screen. But players won't be confused if we use a trophy icon somewhere in the top, another pattern that I've noticed in a couple of games.

**Comment by CelticMinstrel** [comment]
> Another suggestion would be to move language and it's dialog inside the Preferences dialog, making it a tab.

I think it's probably better for language to be visible right on the main screen, in case auto-detecting the user's language didn't work for some reason.

> I wonder if in addition to the global scope, would it be good to have an "Achievements" menu inside the campaign itself, which shows the Achievements from that campaign only.

It sounds like a good idea, but how does the game know which achievements pertain to which campaign? I don't believe there is anything in the achievement definitions to provide that link.

**Comment by babaissarkar** [comment]
> It sounds like a good idea, but how does the game know which achievements pertain to which campaign? I don't believe there is anything in the achievement definitions to provide that link.

This give me hope that it should be doable :

![Screenshot from 2024-04-17 10-47-17](https://github.com/wesnoth/wesnoth/assets/8469888/881f5f38-4087-441f-a103-b596093f1b5d)

Probably this should be discussed in a separate issue, since it's unrelated.

**Comment by Wedge009** [comment]
> Another suggestion would be to move language and it's (sic) dialog inside the Preferences dialog, making it a tab.

I feel language selection should be one of the obvious things on the main menu and not hidden away in another dialogue. If I understand correctly, English isn't your native language, is it? Wouldn't you feel like language is something you'd want to change first?

**Comment by knyghtmare** [comment]
> Preferences - could maybe be moved to the bottom and given a smaller button with a gear icon?

Would certainly be a good design decision 

> Achievements - would prefer to keep, just so players see it and know achievements are a thing that exists

Yeah, should stay as is

> Community - would like to keep to draw players' attention to the community links, but I don't feel strongly about it

Maybe you can place it somewhere else on the screen (upper left area maybe) with a "community icon"?

> Credits - could be added to a unified About dialog instead of having its own button

I think credits can be moved to this "About" dialog

> Quit - somewhat wondering if it could have its own smaller X button elsewhere on the screen, but I'm not sure if that would make it too hard to notice

> Hmm, I would vote to keep as is.

**Comment by knyghtmare** [comment]
My bad. I mis-clicked! Sorry

**Comment by babaissarkar** [comment]
> I feel language selection should be one of the obvious things on the main menu and not hidden away in another dialogue. If I understand correctly, English isn't your native language, is it? Wouldn't you feel like language is something you'd want to change first?

Linux apps usually auto-detect language from the locale. So far Wesnoth is probably the only game I saw with an obvious language selection dialog. That said, I'm find with keeping it in the main screen if others think it's a better idea.

**Comment by babaissarkar** [comment]
Besides, the language selection button is small, so it's not really the one that's creating the main screen clutter.

**Comment by babaissarkar** [comment]
Another idea : 
Keep the language button on the main screen, but also unify the language dialog into the preferences, so that clicking on it actually opens the preferences dialog with the correct tab selected. This has the benefit that players will also be able to change other preferences and language in one go, and the button also remains, so they can use that as well.
The same can be done for the "i" button, it stays, but we move credits and community to About dialog, add the "About" button, and make sure that clicking both "i" and "About" does the same things.
That's a bit of duplication, but since removing them is not an option, this might work.

**Comment by Wedge009** [comment]
I find desired language doesn't always match locale, but fair point. The idea of the buttons as a 'short-cut' into the preferences dialogue is a reasonable option.

**Comment by CelticMinstrel** [comment]
> This give me hope that it should be doable

Those aren't campaigns though. They just happen to have the same names as the campaigns, to make the illusion of a connection.

**Comment by Wedge009** [comment]
> Linux apps usually auto-detect language from the locale...

Revisiting this point, even if automatic language selection is a convention on Linux, Wesnoth is cross-platform. As it stands, it looks like Wesnoth doesn't handle this well (or at all) anyway, given #2558.

**Comment by babaissarkar** [comment]
A very crude demo of the about dialog (modified Build Info dialog) :
![Screenshot from 2024-04-22 20-08-41](https://github.com/wesnoth/wesnoth/assets/8469888/a1bd7ce0-4762-4f34-b018-e4fe97131316)
![Screenshot from 2024-04-22 20-08-34](https://github.com/wesnoth/wesnoth/assets/8469888/08d5f372-d91e-45fa-91ad-1aef1afde9dd)
![Screenshot from 2024-04-22 20-08-26](https://github.com/wesnoth/wesnoth/assets/8469888/f0d9e13a-0df7-4214-8379-54fed9c62a3c)
![Screenshot from 2024-04-22 20-08-17](https://github.com/wesnoth/wesnoth/assets/8469888/5a35263a-1cbb-4b97-bcb6-e1d72ebdc640)

**Comment by CelticMinstrel** [comment]
Does it really make sense for Community to be a button now? As opposed to merging it in with the General tab?

**Comment by babaissarkar** [comment]
> Does it really make sense for Community to be a button now? As opposed to merging it in with the General tab?

Can you elaborate on how do you want them to be added? As links or as buttons, or something else?

**Comment by CelticMinstrel** [comment]
I'm not sure, to be honest. Obviously they were buttons in the previous community dialog. Links might fit better in this context, but there's also the possibility of using small icon buttons.

**Comment by Wedge009** [comment]
> A very crude demo of the about dialog (modified Build Info dialog) 

Are you planning to update the dialogue (it looks pretty good to me so far)? I just noticed #1655 might benefit from this work.

**Comment by Pentarctagon** [comment]
Thinking about it some more, I would prefer that the Community button remain on the title screen so that it's much more visible and players are more likely to go to the various community links.

**Comment by babaissarkar** [comment]
> Thinking about it some more, I would prefer that the Community button remain on the title screen so that it's much more visible and players are more likely to go to the various community links.

That mean only one button gets changed : Credits > About and the buttons numbers don't reduce.
Perhaps an icon, as suggested [here](https://github.com/wesnoth/wesnoth/issues/8768#issuecomment-2059820400).

I wonder if it's possible to merge Campaign, Multiplayer and Load into one single window that allows the player to play SP or MP or load an existing game, though it's a dramatic change so people might disagree.

**Comment by babaissarkar** [comment]
Here's how supertux does it : (Note that some of the menu items are same as ours)
![Screenshot from 2024-04-23 10-03-04](https://github.com/wesnoth/wesnoth/assets/8469888/56051df2-b4e8-42c6-92a3-05e3890a3c7e)

I like the fact that the main menu is highly decorated in their style (the childish, icy font) and it is very similar to their logo, probably the same font. It stays consistent to their theme.

The current wesnoth title screen does have the fantasy feel, but only at the logo, but not in the buttons themselves. They look like application buttons, which is good for in game UI, but not for the title screen, which should be more decorated.

**Comment by Pentarctagon** [comment]
I thought the credits were going into what's currently the build info dialog?

**Comment by babaissarkar** [comment]
Yes. I had thoughts about an "About" button in place of "Credits", but
perhaps just using the "i" would be enough. Perhaps "i" can be made a bit
bigger so it's obvious.

**Comment by babaissarkar** [comment]
Here's the idea :
![Screenshot from 2024-04-23 10-41-23](https://github.com/wesnoth/wesnoth/assets/8469888/5a9cebb0-949e-43e5-9973-2cd6b4ce605d)

(Is the version number necessary in the title screen? Will take a lot of space. Similarly I used "About Wesnoth" instead of "About The Battle for Wesnoth", that'd be even bigger.)

**Comment by Pentarctagon** [comment]
Probably it can just be "About"? You already know the Wesnoth part because, well, you're currently playing Wesnoth, and the version number is already displayed in the window title and next to the current "i" button:
![Screenshot from 2024-04-23 00-25-23](https://github.com/wesnoth/wesnoth/assets/3030250/c7b1a959-0c81-4341-84ba-6303da0424fc)

**Comment by babaissarkar** [comment]
![Screenshot from 2024-04-23 10-58-31](https://github.com/wesnoth/wesnoth/assets/8469888/dda2aeb5-7a79-4693-b68e-0253d03e1a46)

**Comment by CelticMinstrel** [comment]
> The current wesnoth title screen does have the fantasy feel, but only at the logo, but not in the buttons themselves.

I think the title uses Oldania, which is also distributed with the game, so you could try using that for the main menu buttons too.

**Comment by babaissarkar** [comment]
> I think the title uses Oldania, which is also distributed with the game, so you could try using that for the main menu buttons too. It wouldn't replicate the outline/emboss effect, but it would have the same letter shapes.

Yes, thinking on a design. From https://wiki.wesnoth.org/ImageLocalization, the wesnoth logo at the top uses Celtic Garamond.

**Comment by babaissarkar** [comment]
I've created a PR on the About dialog (#8803)

**Comment by CelticMinstrel** [comment]
> From https://wiki.wesnoth.org/ImageLocalization, the wesnoth logo at the top uses Celtic Garamond.

Ah, I guess I was wrong… oh well.

**Comment by babaissarkar** [comment]
> > From https://wiki.wesnoth.org/ImageLocalization, the wesnoth logo at the top uses Celtic Garamond.
> 
> Ah, I guess I was wrong… oh well.

No, @Vultraz pointed out that the wiki page is wrong, and outdated, so it's correctly using Oldania now.

**Comment by Wedge009** [comment]
#8803 has improved the situation somewhat - are we keeping this open because there's still more to be done?

**Comment by Wedge009** [comment]
#8953 is further work that improves the situation here, and emphasises the important selections over the others. I suppose this is hanging around from potential changes of some text selections to icons.

**Comment by Wedge009** [comment]
What's the status of this? While the title has been redesigned in 1.19, the _number_ of buttons in the menu column hasn't changed. I don't mind how the title looks but was there a desire to further simplify (eg with the citation of the Super Tux menu)?

**Comment by mentos987** [comment]
I am also fine with it the way it is. 

But I do have a suggestion that would shorten the list and allow for seamless future addition of more buttons:

<img width="467" height="555" alt="Image" src="https://github.com/user-attachments/assets/88af8b66-d86f-4f95-96b7-11804cd01f0d" />

Clicking on the "other" button would replace the entire button list with:

1. Achievements
2. Map Editor
3. Community
4. Credits (If we want to add it back?)
5. Back/Main Menu (Return to standard menu)
6. Quit

Why do I suggest this? Because this way we have room to grow (add more buttons) and because I do not think the average player will be clicking on "Achievements", "Map Editor" or "Community" nearly as much as they will any of the other buttons that exist on the main page.

**Comment by Wedge009** [comment]
I was referring to the Celes theme, not the previous Parchment theme. While Celes essentially has the same buttons, it does highlight the first three. Additionally, those three you say to hide away, at least two of those are stated to keep in the very first post.

I think everyone is going to have different opinions on what should and shouldn't be emphasised. For example, I don't play MP, so having that in a prominent position doesn't help me but I completely understand that's the main draw for many players.

I suppose I'm asking if Celes addresses some or any of what was discussed here or if it needs to be kept open. But I'm not sure it will go anywhere short of a complete and radical overhaul of the menu.

**Comment by babaissarkar** [comment]
@Pentarctagon your thoughts?

@Wedge009 Celes doesn't.

**Comment by Pentarctagon** [comment]
I'd be in favor of closing this issue at this point.
